R/zchunk_L2321.cement_USA.R

```diff
@@ -68,114 +76,496 @@ module_gcam.usa_L2321.cement_USA_DISABLED <- function(command, ...) {
+      # We are only interested in the technology IDs where calibration = output.
+      filter(calibration == "output") %>%
+      select(sector, calibration, supplysector, subsector, technology) %>%
+      distinct ->
+      calibrated_techs_cement_sector_info
+
+    # Combine the cement sector information found above and the stub-technology calibrated
+    # cement production into a single data frame.
+    L2321.StubTechProd_cement_USA %>%
+      left_join_error_no_match(calibrated_techs_cement_sector_info, by = "sector") %>%
+      select(state, sector, calOutputValue, year, region, supplysector, subsector, technology) ->
+      L2321.StubTechProd_cement_USA
+
+    # Add share weight information to the state cement production data frame and format.
+    L2321.StubTechProd_cement_USA %>%
+      mutate(stub.technology = technology) %>%
+      mutate(share.weight.year = year) %>%
+      mutate(subs.share.weight = if_else(calOutputValue > 0, 1, 0)) %>%
+      mutate(tech.share.weight = subs.share.weight) %>%
+      select(region, supplysector, subsector, stub.technology, year, calOutputValue,
+             share.weight.year, subs.share.weight, tech.share.weight) ->
+      L2321.StubTechProd_cement_USA
+
+
+    # Create the coefficients of cement production technologies input table.
     #
+    # Start by creating a data frame of unique sector, fuel, supplysector, subsector,
+    # technology, and minicam.energy.input combinations. This data frame will be used to
+    # add sector information to input-ouput coefficients for state cement production.
+    calibrated_techs %>%
+      select(sector, fuel, supplysector, subsector, technology, minicam.energy.input) %>%
+      distinct ->
+      cement_production_technologies
+
+    # Add cement sector information to the data frame of input-output coefficients of
+    # cement production by state.
+    L1321.IO_GJkg_state_cement_F_Yh %>%
+      left_join_error_no_match(cement_production_technologies, by = c("sector", "fuel")) %>%
+      select(state,fuel, sector, year, value, supplysector, subsector, technology, minicam.energy.input) ->
+      L2321.IO_GJkg_state_cement_F_Yh
+
+    # Interpolate cement production default coefficients to future years.
     #
-    # NOTE: there are 'match' calls in this code. You probably want to use left_join_error_no_match
-    # For more information, see https://github.com/JGCRI/gcamdata/wiki/Name-That-Function
-    # NOTE: This code uses vecpaste
-    # This function can be removed; see https://github.com/JGCRI/gcamdata/wiki/Name-That-Function
-    # NOTE: This code uses repeat_and_add_vector
-    # This function can be removed; see https://github.com/JGCRI/gcamdata/wiki/Name-That-Function
-    # ===================================================
+    # First change format of the the production default coefficients data frame from wide to long.
+    A321.globaltech_coef %>%
+      gather(year, value, matches(YEAR_PATTERN)) %>%
+      mutate(year = as.integer(year)) ->
```

Comment:
Ideally we'll get #821 merged first and you can use `gather_years` here

Reply:
Replace these new lines with call to new `gather_year` function ( #821 ). This also applies to all `gather %>% mutate(year = as.integer(year))` sequences below.